en/22/11.md

```diff
@@ -0,0 +1,296 @@
+      self.adder_proxy(adder_contract_address)
+          .add(Biguint::from(2u32), Biguint::from(2u32))
+          .async_call()
+          .with_callback(self.callbacks().add_callback(&caller))
+          .call_and_exit();
+    }
+
+    #[callback]
+    fn add_callback(
+      &self, 
+      #[call_result] result: ManagedAsyncCallResult<BigUint>,
+      caller: &ManagedAddress
+    ) {
+        match result {
+            ManagedAsyncCallResult::Ok(sum) => {
+                self.sum(caller).set(sum);
+            },
+            ManagedAsyncCallResult::Err(_) => {
+                // this can only fail if the adder contract address is invalid
+                // nothing to revert in case of error
+            },
+        }
+    }
+
+    #[view(getSum)]
+    #[storage_mapper("sum")]
+    fn adder_contract_address(&self) -> SingleValueMapper<ManagedAddress>;
+
+    #[view(getSum)]
+    #[storage_mapper("sum")]
+    fn sum(&self, caller: &ManagedAddress) -> SingleValueMapper<BigUint>;
+
+```
+
+When calling another contract, if both contracts are in the same shard or not, we can make a direct call or a asynchronus call to one of their endpoints. In this context we can assume that the contracts are on diferent shards, reason why we used `async_call`.
+
+Since we are talking about asynchronus call we have no idea when we will recieve a response from the other contract (or if we will), reason why we will attach to this call another type of function defined by us, a callback which when will receive the result he will do something with it. The result is basically a fancy looking option with the result type of the called endpoint as a content or an error which in this case we will just ignore. A callback is just a basic function having the `#[callback]` annotation.
+
+# Put it to the test
+
+Let's set up our contract to read from the CryptoKitties smart contract!
+I've created a storage mapper named `crypto_kitties_sc_address` in which the address of the CryptoKitties contract will be saved. 
+
+1. Under the `feed_and_multiply` endpoint create a callback called `get_kitty_callback` that will take 1 `usize` parameter `zombie_id` (besides the result marked with `#[call_result]` of course). Fill the match just like in the example abuve but by leaving both case block empty for now. Be careful for the return type of the result  `#[call_result] result: ManagedAsyncCallResult<Kitty>` and to import the `Kitty` struct inside the zombiefeeding module (add `use crypto_kitties_proxy::Kitty;` just under `use crate::{storage, zombiefactory};`)
+   
+2. create another endpoint called `feed_on_kitty`. It will take 2 `usize` parameters, `zombie_id` and `kitty_id`.
+
+3. Inside this function get the address of the crypto kitties contract and call the `get_kitty` proxy function  (giving it `kitty_id` as parameter) with asynchronus call and callback on the callback defined before (giving it `zombie_id` as parameter).
```

Comment:
Sorry, but I can't understand what students need to do here -> "with asynchronus call and callback on the callback defined before". Please consider rephrasing/simplifying this step.